lib/course_planner/notifier.ex

```diff
@@ -0,0 +1,33 @@
+defmodule CoursePlanner.Notifier do
+  @moduledoc """
+  Module responsible for notifying users through e-mail with changes
+  """
+  use GenServer
+  alias CoursePlanner.{User, Mailer, Mailer.UserEmail}
+  require Logger
+
+  @spec start_link :: GenServer.start_link
+  def start_link do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @spec notify_user(User, atom()) :: GenServer.cast
+  def notify_user(%User{} = user, notification_type) do
+    GenServer.cast(__MODULE__, {:send, user, notification_type})
+  end
+
+  @spec handle_cast({atom(), User, atom()}, any()) :: {:noreply, any()}
+  def handle_cast({:send, user, notification_type}, state) do
```

Comment:
we should have one `handle_cast` which consumes the messages that don't match the pattern in order to clean the mailbox